typing/ctype.ml

```diff
@@ -1484,13 +1532,17 @@ let subst env level priv abbrev oty params args body =
   abbreviations := abbrev;
   let (params', body') = instance_parameterized_type params body in
   abbreviations := ref Mnil;
+  let old_umode = !umode in
```

Comment:
I was a little worried that `protect_refs` would adversely affect performance, and since we're already catching the only exception that should ever come out of here I decided to do it directly instead.

Reply:
Fair enough. I don't know the perf implications.

Reply:
It's almost certainly premature optimisation but I decided it was easier to just do this than to work out if it could matter.